python/tk_houdini/ui_generation.py

```diff
@@ -134,6 +134,98 @@ def _itemNode(self, parent, label, id):
         node.text = id
         return item
 
+class AppCommandsPanels(AppCommandsUI):
+
+    def __init__(self, engine, commands=None, panel_commands=None):
+        self._panel_commands = panel_commands
+        super(AppCommandsPanels, self).__init__(engine, commands)
+
+        if self._panel_commands is None:
+            self._panel_commands = get_registered_panels(engine)
```

Comment:
Ok, ignore the first part of my previous reply to your comment. Making up hours on Sunday morning... probably not the best idea. I don't know why I made those arguments optional, and I think I'm going to go back and change the way this works because, as you said, the logic is not clear.